docs/better-sandwiches/index.md

```diff
@@ -0,0 +1,290 @@
+hard and fast rule. In this case, it's useful to temporarily use wildcards to
+silence the compiler errors while we fill in the missing logic for
+`Item.toPrice` and `Item.toEmoji`. Still, when the wildcard is meant to be
+temporary, we can give it a more attention-grabbing name so it's less likely to
+become permanent:
+
+```reason
+| Sandwich(_todo_please_use_this_sandwich) => 10.
+```
+
+## Update `Item.toPrice` function
+
+The price of a sandwich should vary depending on the type of sandwich received
+by `Item.toPrice`[^2]:
+
+<<< Item.re#to-price-nested-switch
+
+Actually, there's no need for the nested switch, as OCaml allows you to pattern
+match into nested structures:
+
+<<< Item.re#to-price-flat
+
+## Combining branches in switch
+
+Madame Jellobutter decides to upgrade her portabellos from canned mushrooms to
+artisanal mushrooms lovingly grown by mustachioed hipsters on a local farm,
+which increases the price of portabello sandwiches to $10. Now portabello and
+ham sandwiches are the same price, so you update the `Item.toPrice` function
+accordingly:
+
+<<< Item.re#to-price-same-price{3}
+
+We can combine the portabello and ham branches because they return the same
+value:
+
+<<< Item.re#to-price-combine{3-4}
+
+Since the outer constructor `Sandwich` is the same in both branches, we can
+combine them into a single branch:
+
+<<< Item.re#to-price{3}
+
+The power of pattern matching gives us both clarity and brevity. This is the
+OCaml Way!
+
+## Update `Item.toEmoji` function
+
+`Item.toEmoji` should now return a combination of two emojis when it gets a
+`Sandwich(_)` argument:
+
+| Sandwich | Emoji |
+| -------- | ----- |
+| Portabello | 🥪(🍄) |
+| Ham | 🥪(🐷) |
+| Unicorn | 🥪(🦄) |
+
+We can implement this using `Js.Array.joinWith`:
+
+<<< Item.re#to-emoji-joinwith{3-13}
+
+This touches on the issue of coding style for switch expressions. When possible,
+put shorter branches before longer branches:
+
+<<< Item.re#to-emoji-short-branch-first
+
+## `j` quoted string literal
+
+The approach using `Js.Array.joinWith` works, but readability could be improved
+by using a [quoted string literal with the quoted string identifier
+`j`](https://melange.re/v2.2.0/communicate-with-javascript/#strings):
+
+<<< Item.re#to-emoji-j-string-literal{5-13}
+
+`j` quoted string literals are similar to [template literals in
+JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals)
+except that they can only accept variables, not arbitrary expressions. Variables
+that appear in `j` quoted string literals are implicitly converted to strings,
+so they are not type safe and thus unsuitable for production code. See the
+[exercises at the end of this chapter](#exercises) for more details.
+
+Also note that unlike switch expressions, the `fun` syntax does not accept
+multi-line expressions in branches unless you add `{}` around them.
+
+## `Printf.sprintf`
```

Comment:
I wonder if I should add a warning about how much using `Printf.sprintf` adds to the bundle size?

Reply:
Yes, that'd be great I think.

Reply:
Added an info box about how much `Printf` adds to bundle size in https://github.com/melange-re/melange-for-react-devs/pull/13/commits/c3f6d8db0d132a10a8da5ad278b5b717e1aded91